RenderStatePackager/src/RenderStatePackager.cpp

```diff
@@ -33,217 +33,314 @@ namespace Diligent
+
+                        std::unique_lock<std::mutex> Lock{m_LockShaders};
+                        m_Shaders.emplace(ShaderCI.Desc.Name, pShader);
+                    }
+                    catch (...)
+                    {
+                        Exceptions.push_back(std::current_exception());
+                    }
+                });
+
+                Tasks.push_back(pTask);
+            }
 
-            ResourceDesc.pTS = FindShader(pResourceDescRSN->pTSName);
+            for (Uint32 RenderPassID = 0; RenderPassID < ParserInfo.RenderPassCount; RenderPassID++)
+            {
+                auto pTask = EnqueueAsyncWork(m_pThreadPool, [RenderPassID, pNotationParser, this, &Exceptions](Uint32 ThreadId) {
+                    try
+                    {
+                        RenderPassDesc             RPDesc = *pNotationParser->GetRenderPassByIndex(RenderPassID);
+                        RefCntAutoPtr<IRenderPass> pRenderPass;
+                        m_pDevice->CreateRenderPass(RPDesc, &pRenderPass);
+                        if (!pRenderPass)
+                            LOG_ERROR_AND_THROW("Failed to create render pass '", RPDesc.Name, "'.");
+
+                        std::unique_lock<std::mutex> Lock{m_LockRenderPasses};
+                        m_RenderPasses.emplace(RPDesc.Name, pRenderPass);
+                    }
+                    catch (...)
+                    {
+                        Exceptions.push_back(std::current_exception());
+                    }
+                });
+                Tasks.push_back(pTask);
+            }
 
-            PipelineStateArchiveInfo ArchiveInfo = {};
-            ArchiveInfo.DeviceFlags              = m_DeviceBits;
-            if (!pArchive->AddTilePipelineState(ResourceDesc, ArchiveInfo))
-                LOG_ERROR_AND_THROW("Failed to archive tile pipeline '", ResourceDesc.PSODesc.Name, "'.");
+            for (Uint32 SignatureID = 0; SignatureID < ParserInfo.ResourceSignatureCount; SignatureID++)
+            {
+                auto pTask = EnqueueAsyncWork(m_pThreadPool, [SignatureID, pNotationParser, this, &Exceptions](Uint32 ThreadId) {
+                    try
+                    {
+                        PipelineResourceSignatureDesc             SignDesc = *pNotationParser->GetResourceSignatureByIndex(SignatureID);
+                        RefCntAutoPtr<IPipelineResourceSignature> pSignature;
+                        m_pDevice->CreatePipelineResourceSignature(SignDesc, m_DeviceBits, &pSignature);
+                        if (!pSignature)
+                            LOG_ERROR_AND_THROW("Failed to create resource signature '", SignDesc.Name, "'.");
+
+                        std::unique_lock<std::mutex> Lock{m_LockResourceSignatures};
+                        m_ResourceSignatures.emplace(SignDesc.Name, pSignature);
+                    }
+                    catch (...)
+                    {
+                        Exceptions.push_back(std::current_exception());
```

Comment:
Exceptions used without mutex - resize to m_RSNParsers.size() to avoid the problem